### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by mplachta**
@Schibsted-Tech-Polska/sa

### CODE REVIEW

File: src/Stp/SndApi/Common/Client.php [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    protected $apiUrl = self::BASE_URL;
+
+    /**
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * @param $apiSecret string
+     * @param $publicationId string
```
Comment (filipgolonka) [inline src/Stp/SndApi/Common/Client.php]: shouldn't it be `@param string $apiSecret`?

File: src/Stp/SndApi/Common/Validator/PublicationIdValidator.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        'sa',
+        'fvn',
+        'bt',
+        'ap',
+    ];
+
+    private $publicationId;
+
+    /**
+     * @param $publicationId
```
Comment (filipgolonka) [inline src/Stp/SndApi/Common/Validator/PublicationIdValidator.php]: missing variable type

File: src/Stp/SndApi/News/Command/ArticleCommand.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+use Stp\SndApi\Common\Command\BaseCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class ArticleCommand extends BaseCommand
+{
+    use NewsClientCommandTrait;
+
+    protected function configure()
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Command/ArticleCommand.php]: missing PHPDoc

File: src/Stp/SndApi/News/Command/ArticleCommand.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        $this
+            ->setName('news:article:id')
+            ->addArgument(
+                'id',
+                InputArgument::REQUIRED,
+                'Article ID'
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Command/ArticleCommand.php:33]: missing PHPDoc

File: src/Stp/SndApi/News/Command/ArticlesBySectionIdCommand.php [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+use Stp\SndApi\Common\Command\BaseCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class ArticlesBySectionIdCommand extends BaseCommand
+{
+    use NewsClientCommandTrait;
+
+    protected function configure()
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Command/ArticlesBySectionIdCommand.php]: missing PHPDoc

File: src/Stp/SndApi/News/Command/ArticlesBySectionIdCommand.php [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+                'Articles list method (desked, auto)'
+            )
+            ->addArgument(
+                'parameters',
+                InputArgument::IS_ARRAY | InputArgument::OPTIONAL,
+                'List of parameters i.e. limit=50'
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Command/ArticlesBySectionIdCommand.php:45]: missing PHPDoc

File: src/Stp/SndApi/News/Client.php [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+
+        $url = rtrim($url, '?&');
+
+        $response = $this->apiGet($url);
+
+        return $response->json();
+    }
+
+    /**
+     * @param $contentId
```
Comment (wsromek) [inline src/Stp/SndApi/News/Client.php]: where is type?

File: src/Stp/SndApi/News/Client.php [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+            throw new InvalidMethodParametersException(
+                sprintf('Invalid parameters used for "%s" method', $method)
+            );
+        }
+
+        $url = sprintf('/publication/{publicationId}/sections/%d/%s?', $sectionId, $method);
+
+        foreach ($parameters as $key => $value) {
+            $url .= sprintf('%s=%s&', urlencode($key), urlencode($value));
+        }
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Client.php]: use https://php.net/http_build_query

File: src/Stp/SndApi/News/Command/SearchByInstanceCommand.php [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+use Stp\SndApi\Common\Command\BaseCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class SearchByInstanceCommand extends BaseCommand
+{
+    use NewsClientCommandTrait;
+
+    protected function configure()
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Command/SearchByInstanceCommand.php]: missing PHPDoc

File: src/Stp/SndApi/News/Command/SearchByInstanceCommand.php [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+                'Content ID'
+            )
+            ->addArgument(
+                'contentType',
+                InputArgument::REQUIRED,
+                'Content type (article, person)'
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Command/SearchByInstanceCommand.php:39]: missing PHPDoc

File: src/Stp/SndApi/News/Command/SearchCollectionCommand.php [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+use Stp\SndApi\Common\Command\BaseCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class SearchCollectionCommand extends BaseCommand
+{
+    use NewsClientCommandTrait;
+
+    protected function configure()
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Command/SearchCollectionCommand.php]: missing PHPDoc

File: src/Stp/SndApi/News/Command/SearchCollectionCommand.php [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+        $this
+            ->setName('news:search:collection')
+            ->addArgument(
+                'ids',
+                InputArgument::REQUIRED | InputArgument::IS_ARRAY,
+                'Content IDs'
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Command/SearchCollectionCommand.php:33]: missing PHPDoc

File: src/Stp/SndApi/News/Command/SectionByIdCommand.php [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+use Stp\SndApi\Common\Command\BaseCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class SectionByIdCommand extends BaseCommand
+{
+    use NewsClientCommandTrait;
+
+    protected function configure()
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Command/SectionByIdCommand.php]: missing PHPDoc

File: src/Stp/SndApi/News/Command/SectionByIdCommand.php [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+        $this
+            ->setName('news:section:id')
+            ->addArgument(
+                'id',
+                InputArgument::REQUIRED,
+                'Section ID'
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Command/SectionByIdCommand.php:33]: missing PHPDoc

File: src/Stp/SndApi/News/Command/SectionByUniqueNameCommand.php [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+use Stp\SndApi\Common\Command\BaseCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class SectionByUniqueNameCommand extends BaseCommand
+{
+    use NewsClientCommandTrait;
+
+    protected function configure()
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Command/SectionByUniqueNameCommand.php]: missing PHPDoc

File: src/Stp/SndApi/News/Command/SectionByUniqueNameCommand.php [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+        $this
+            ->setName('news:section:uniquename')
+            ->addArgument(
+                'name',
+                InputArgument::REQUIRED,
+                'Unique section name'
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Command/SectionByUniqueNameCommand.php:33]: missing PHPDoc

File: src/Stp/SndApi/News/Command/SectionsListCommand.php [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+use SoapBox\Formatter\Formatter;
+use Stp\SndApi\Common\Command\BaseCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class SectionsListCommand extends BaseCommand
+{
+    use NewsClientCommandTrait;
+
+    protected function configure()
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Command/SectionsListCommand.php]: missing PHPDoc

File: src/Stp/SndApi/News/Command/SectionsListCommand.php [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        parent::configure();
+
+        $this
+            ->setName('news:sections:list');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Command/SectionsListCommand.php:27]: missing PHPDoc

File: src/Stp/SndApi/News/Command/ServiceDocumentCommand.php [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+use SoapBox\Formatter\Formatter;
+use Stp\SndApi\Common\Command\BaseCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class ServiceDocumentCommand extends BaseCommand
+{
+    use NewsClientCommandTrait;
+
+    protected function configure()
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Command/ServiceDocumentCommand.php]: missing PHPDoc

File: src/Stp/SndApi/News/Command/ServiceDocumentCommand.php [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        parent::configure();
+
+        $this
+            ->setName('news:servicedocument');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Command/ServiceDocumentCommand.php:27]: missing PHPDoc

File: src/Stp/SndApi/News/Command/SubsectionsListCommand.php [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+use Stp\SndApi\Common\Command\BaseCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class SubsectionsListCommand extends BaseCommand
+{
+    use NewsClientCommandTrait;
+
+    protected function configure()
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Command/SubsectionsListCommand.php]: missing PHPDoc

File: src/Stp/SndApi/News/Command/SubsectionsListCommand.php [hunk 22]
```diff
@@ -0,0 +1,39 @@
...
+        $this
+            ->setName('news:subsections:list')
+            ->addArgument(
+                'sectionId',
+                InputArgument::REQUIRED,
+                'Section ID'
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Command/SubsectionsListCommand.php:33]: missing PHPDoc

File: src/Stp/SndApi/News/Command/ArticlesBySectionIdCommand.php [hunk 23]
```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Stp\SndApi\News\Command;
+
+use SoapBox\Formatter\Formatter;
```
Comment (wsromek) [inline src/Stp/SndApi/News/Command/ArticlesBySectionIdCommand.php]: Maybe use Symfony\Component\Yaml\Yaml; ?

File: src/Stp/SndApi/News/Command/ArticlesBySectionIdCommand.php [hunk 24]
```diff
@@ -0,0 +1,58 @@
...
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $newsClient = $this->getNewsClient($input);
+
+        $sectionId = $input->getArgument('id');
+        $method = $input->getArgument('method');
+        $parameterStrings = $input->getArgument('parameters');
+        $parameters = [];
+
+        foreach ($parameterStrings as $parameter) {
```
Comment (wsromek) [inline src/Stp/SndApi/News/Command/ArticlesBySectionIdCommand.php]: could it be more readable?

File: src/Stp/SndApi/News/Test/Command/ApiCommandTest.php [hunk 25]
```diff
@@ -0,0 +1,91 @@
...
+use Stp\SndApi\News\Command\SectionByUniqueNameCommand;
+use Stp\SndApi\News\Command\SectionsListCommand;
+use Stp\SndApi\News\Command\ServiceDocumentCommand;
+use Stp\SndApi\News\Command\SubsectionsListCommand;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Tester\CommandTester;
+
+class ApiCommandTest extends \PHPUnit_Framework_TestCase
+{
+    public function initCommand($command, $commandClass, $expectedMethod)
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Test/Command/ApiCommandTest.php]: private

File: src/Stp/SndApi/News/Validator/ArticleListMethodValidator.php [hunk 26]
```diff
@@ -0,0 +1,28 @@
...
+class ArticleListMethodValidator implements ValidatorInterface
+{
+    private $allowedMethods = [
+        'desked',
+        'auto'
+    ];
+
+    private $method;
+
+    public function __construct($method)
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Validator/ArticleListMethodValidator.php:19]: missing PHPDoc

File: src/Stp/SndApi/News/Validator/ArticlesListParametersValidator.php [hunk 27]
```diff
@@ -0,0 +1,48 @@
...
+            'since',
+            'until'
+        ]
+    ];
+
+    private $method;
+
+    private $parameters;
+
+    public function __construct($method, $parameters)
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Validator/ArticlesListParametersValidator.php:31]: missing PHPDoc

File: src/Stp/SndApi/News/Validator/ContentTypeValidator.php [hunk 28]
```diff
@@ -0,0 +1,28 @@
...
+class ContentTypeValidator implements ValidatorInterface
+{
+    private $allowedArticleContentType = [
+        'article',
+        'person'
+    ];
+
+    private $contentType;
+
+    public function __construct($contentType)
```
Comment (filipgolonka) [inline src/Stp/SndApi/News/Validator/ContentTypeValidator.php:19]: missing PHPDoc

File: src/Stp/SndApi/Common/Client.php (hunk 1 above)
Comment (mplachta) [inline src/Stp/SndApi/Common/Client.php]: :+1:

File: src/Stp/SndApi/Common/Validator/PublicationIdValidator.php (hunk 2 above)
Comment (mplachta) [inline src/Stp/SndApi/Common/Validator/PublicationIdValidator.php]: :+1:

File: src/Stp/SndApi/News/Command/ArticlesBySectionIdCommand.php (hunk 5 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Command/ArticlesBySectionIdCommand.php]: :-1:

File: src/Stp/SndApi/News/Command/ArticleCommand.php (hunk 4 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Command/ArticleCommand.php:33]: :+1:

File: src/Stp/SndApi/Common/Client.php [hunk 29]
```diff
@@ -0,0 +1,174 @@
...
+        $request->addHeader('Accept-Charset', 'UTF-8');
+    }
+
+    /**
+     * @param $url
+     * @return array|Response|null
+     */
+    protected function apiGet($url)
+    {
+        if (0 === preg_match('/^\//', $url)) {
```
Comment (rafalkanski) [inline src/Stp/SndApi/Common/Client.php]: $url[0] !== '/' won't be faster?

File: src/Stp/SndApi/News/Command/ArticlesBySectionIdCommand.php (hunk 6 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Command/ArticlesBySectionIdCommand.php:45]: :+1:

File: src/Stp/SndApi/Common/Client.php [hunk 30]
```diff
@@ -0,0 +1,174 @@
...
+
+    /**
+     * @param string $publicationId
+     *
+     * @throws InvalidPublicationIdException
+     * @SuppressWarnings(PHPMD.LongVariable)
+     */
+    public function setPublicationId($publicationId)
+    {
+        $publicationIdValidator = new PublicationIdValidator($publicationId);
```
Comment (rafalkanski) [inline src/Stp/SndApi/Common/Client.php]: i really like it :+1:

File: src/Stp/SndApi/News/Client.php (hunk 8 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Client.php]: :+1:

File: src/Stp/SndApi/News/Command/SearchByInstanceCommand.php (hunk 9 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Command/SearchByInstanceCommand.php]: :-1:

File: src/Stp/SndApi/News/Command/SearchByInstanceCommand.php (hunk 10 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Command/SearchByInstanceCommand.php:39]: :+1:

File: src/Stp/SndApi/News/Command/SearchCollectionCommand.php (hunk 11 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Command/SearchCollectionCommand.php]: :-1:

File: src/Stp/SndApi/News/Command/SearchCollectionCommand.php (hunk 12 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Command/SearchCollectionCommand.php:33]: :+1:

File: src/Stp/SndApi/News/Command/SectionByIdCommand.php (hunk 13 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Command/SectionByIdCommand.php]: :-1:

File: src/Stp/SndApi/News/Command/SectionByIdCommand.php (hunk 14 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Command/SectionByIdCommand.php:33]: :+1:

File: src/Stp/SndApi/News/Command/SectionByUniqueNameCommand.php (hunk 15 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Command/SectionByUniqueNameCommand.php]: :-1:

File: src/Stp/SndApi/News/Command/SectionByUniqueNameCommand.php (hunk 16 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Command/SectionByUniqueNameCommand.php:33]: :+1:

File: src/Stp/SndApi/News/Command/SectionsListCommand.php (hunk 17 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Command/SectionsListCommand.php]: :-1:

File: src/Stp/SndApi/News/Command/SectionsListCommand.php (hunk 18 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Command/SectionsListCommand.php:27]: :+1:

File: src/Stp/SndApi/News/Command/ServiceDocumentCommand.php (hunk 19 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Command/ServiceDocumentCommand.php]: :-1:

File: src/Stp/SndApi/News/Command/ServiceDocumentCommand.php (hunk 20 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Command/ServiceDocumentCommand.php:27]: :+1:

File: src/Stp/SndApi/News/Command/SubsectionsListCommand.php (hunk 21 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Command/SubsectionsListCommand.php]: :-1:

File: src/Stp/SndApi/News/Command/SubsectionsListCommand.php (hunk 22 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Command/SubsectionsListCommand.php:33]: :+1:

File: src/Stp/SndApi/News/Command/ArticlesBySectionIdCommand.php (hunk 23 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Command/ArticlesBySectionIdCommand.php]: :-1:

File: src/Stp/SndApi/News/Command/ArticlesBySectionIdCommand.php (hunk 24 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Command/ArticlesBySectionIdCommand.php]: :-1:

File: src/Stp/SndApi/News/Test/Command/ApiCommandTest.php (hunk 25 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Test/Command/ApiCommandTest.php]: :+1:

File: src/Stp/SndApi/News/Validator/ArticleListMethodValidator.php (hunk 26 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Validator/ArticleListMethodValidator.php:19]: :+1:

File: src/Stp/SndApi/News/Validator/ArticlesListParametersValidator.php (hunk 27 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Validator/ArticlesListParametersValidator.php:31]: :+1:

File: src/Stp/SndApi/News/Validator/ContentTypeValidator.php (hunk 28 above)
Comment (mplachta) [inline src/Stp/SndApi/News/Validator/ContentTypeValidator.php:19]: :+1:

File: src/Stp/SndApi/Common/Client.php (hunk 29 above)
Comment (mplachta) [inline src/Stp/SndApi/Common/Client.php]: :-1:

**Comment by mplachta** [comment]
@filipgolonka @rafalkanski @wsromek update

**Comment by filipgolonka** [comment]
but reset/rebase needed :)

**Comment by rafalkanski** [comment]
:+1:

**Comment by wsromek** [comment]
as above :+1:

**Pull request merged**